dask/dataframe/io/tests/test_parquet.py

```diff
@@ -1089,6 +1089,25 @@ def test_partition_on(tmpdir, engine):
         assert set(df.b[df.a2 == val]) == set(out.b[out.a2 == val])
 
 
+def test_partition_on_duplicates(tmpdir, engine):
+    tmpdir = str(tmpdir)
+    df = pd.DataFrame(
+        {
+            "a1": np.random.choice(["A", "B", "C"], size=100),
+            "a2": np.random.choice(["X", "Y", "Z"], size=100),
+            "data": np.random.random(size=100),
+        }
+    )
+    d = dd.from_pandas(df, npartitions=2)
+
+    for _ in range(2):
+        d.to_parquet(tmpdir, partition_on=["a1", "a2"], engine=engine)
+
+    out = dd.read_parquet(tmpdir, engine=engine).compute()
+
+    assert len(df) == len(out)
```

Comment:
Please add a commend referencing your issue, as this assert is what would have failed in the previous version.

Reply:
Added a reference.